packages/sftp-client/index.js

```diff
@@ -0,0 +1,267 @@
+'use strict';
+
+const get = require('lodash.get');
+const KMS = require('@cumulus/aws-client/KMS');
+const log = require('@cumulus/common/log');
+const path = require('path');
+const S3 = require('@cumulus/aws-client/S3');
+const { Client } = require('ssh2');
+const { lookupMimeType } = require('@cumulus/common/util');
+const { PassThrough, Readable } = require('stream');
+const { S3KeyPairProvider } = require('@cumulus/common/key-pair-provider');
+
+const decrypt = async (ciphertext) => {
+  try {
+    return await KMS.decryptBase64String(ciphertext);
+  } catch (_) {
+    return S3KeyPairProvider.decrypt(ciphertext);
+  }
+};
+
+class SftpClient {
+  constructor(config) {
+    this.config = config;
+    this.connected = false;
+
+    if (get(config, 'encrypted', false) === false) {
+      this.plaintextUsername = config.username;
+      this.plaintextPassword = config.password;
+    }
+  }
+
+  async getUsername() {
```

Comment:
This might be overkill, but any way to share these functions with the base provider client from above using composition or inheritance?

Reply:
I thought about it and came down on the "overkill" side of the line, especially since they're in two separate npm packages.

Reply:
I think it could make some sense to put both of the FTP clients into a single package, but not blocking for this PR